cc/dual_net/inference_cache.cc

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "cc/dual_net/inference_cache.h"
+
+namespace minigo {
+
+InferenceCache::CompressedFeatures InferenceCache::CompressFeatures(
+    const DualNet::BoardFeatures& features) {
+  CompressedFeatures result;
+  auto num_full_u64s = (features.size() + 63) / 64 - 1;
+  const auto* src = features.data();
+  for (size_t i = 0; i < num_full_u64s; ++i) {
+    uint64_t bits = 0;
+    for (uint64_t j = 0; j < 64; ++j) {
+      bits |= static_cast<uint64_t>(*src++ != 0) << j;
+    }
+    result[i] = bits;
+  }
+  uint64_t bits = 0;
+  for (uint64_t j = 0; src != features.end(); ++j) {
```

Comment:
is it possible to put `j < 64 && src != features.end()` in the above loop to cut down on the code repitition?